berserker-mqtt/src/main/java/io/smartcat/berserker/mqtt/configuration/MqttConfiguration.java

```diff
@@ -0,0 +1,156 @@
+
+import io.smartcat.berserker.api.Worker;
+import io.smartcat.berserker.configuration.ConfigurationParseException;
+import io.smartcat.berserker.configuration.WorkerConfiguration;
+import io.smartcat.berserker.mqtt.worker.AsyncMqttWorker;
+import io.smartcat.berserker.mqtt.worker.SyncMqttWorker;
+
+/**
+ * Configuration for MQTT worker.
+ */
+public class MqttConfiguration implements WorkerConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MqttConfiguration.class);
+
+    private static final String ASYNC = "async";
+    private static final String BROKER_URL = "broker-url";
+    private static final String CLIENT_ID = "client-id";
+
+    private static final String MAX_INFLIGHT = "max-inflight";
+    private static final String CLEAN_SESSION = "clean-session";
+    private static final String CONNECTION_TIMEOUT = "connection-timeout";
+    private static final String MQTT_VERSION = "mqtt-version";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+
+    private static final String MQTT_VERSION_3_1_1 = "3.1.1";
+    private static final String MQTT_VERSION_3_1 = "3.1";
+
+    @Override
+    public String getName() {
+        return "MQTT";
+    }
+
+    @Override
+    public Worker<?> getWorker(Map<String, Object> configuration) throws ConfigurationParseException {
+        Boolean async = (Boolean) configuration.get(ASYNC);
+        String brokerUrl = (String) configuration.get(BROKER_URL);
+        String clientId = (String) configuration.get(CLIENT_ID);
+        Integer maxInflight = (Integer) configuration.get(MAX_INFLIGHT);
+        Boolean cleanSession = (Boolean) configuration.get(CLEAN_SESSION);
+        Integer connectionTimeout = (Integer) configuration.get(CONNECTION_TIMEOUT);
+        String mqttVersion = (String) configuration.get(MQTT_VERSION);
+        String username = (String) configuration.get(USERNAME);
+        String password = (String) configuration.get(PASSWORD);
+
+        validateField(brokerUrl, BROKER_URL);
+        validateField(clientId, CLIENT_ID);
+        int calculatedMaxInflight = calculateMaxInflight(maxInflight);
+        boolean calculatedCleanSession = calculateCleanSession(cleanSession);
+        int calculatedConnectionTimeout = calculateConnectionTimeout(connectionTimeout);
+        int calculatedmqttVersion = calculateMqttVersion(mqttVersion);
+
+        MqttConnectOptions connectOptions = new MqttConnectOptions();
+        connectOptions.setMaxInflight(calculatedMaxInflight);
+        connectOptions.setCleanSession(calculatedCleanSession);
+        connectOptions.setConnectionTimeout(calculatedConnectionTimeout);
+        connectOptions.setMqttVersion(calculatedmqttVersion);
+        if (username != null) {
+            connectOptions.setUserName(username);
+        }
+        if (password != null) {
+            connectOptions.setPassword(password.toCharArray());
+        }
+        if (async == null) {
```

Comment:
Unconditional else is unnecessary since there is no difference in initializing the client. Also the `MqttClient` should be wrapped within Sync or Async implementations.

Reply:
Unconditional else is there just for the log message purpose, but I agree, it can be simplified.

About `MqttClient` wrapping, I'm not sure I understand. You mean to create proxy and use it within Async and Sync implementations in order to simplify it and extract common code to some `BaseMqttWorker` class?